src/DesktopHeader.jsx

```diff
@@ -19,76 +19,50 @@ class DesktopHeader extends React.Component {
     super(props);
   }
 
-  renderMainMenu() {
-    const { mainMenu } = this.props;
-
-    // Nodes are accepted as a prop
-    if (!Array.isArray(mainMenu)) {
-      return mainMenu;
-    }
-
-    return mainMenu.map((menuItem) => {
-      const {
-        type,
-        href,
-        content,
-        submenuContent,
-      } = menuItem;
-
-      if (type === 'item') {
-        return (
-          <a key={`${type}-${content}`} className="nav-link" href={href}>{content}</a>
-        );
-      }
-
-      return (
-        <Menu key={`${type}-${content}`} tag="div" className="nav-item" respondToPointerEvents>
-          <MenuTrigger tag="a" className="nav-link d-inline-flex align-items-center" href={href}>
-            {content} <CaretIcon role="img" aria-hidden focusable="false" />
-          </MenuTrigger>
-          <MenuContent className="pin-left pin-right shadow py-2">
-            {submenuContent}
-          </MenuContent>
-        </Menu>
-      );
-    });
-  }
+  userMenuWithUsername() {
```

Comment:
I see a lot of code repetition in this file. Can we have a parametrized `userMenu` function, please? This will help avoid inconsistencies in future if any.

something like

```
userMenu(usernameOrName) {

..remaining code... 

        <MenuTrigger
          tag="button"
          aria-label={intl.formatMessage(messages['header.label.account.menu.using.usernameOrName'], { [usernameOrName]: usernameOrName })}
          className="btn btn-outline-primary d-inline-flex align-items-center pl-2 pr-3"
        >
          <Avatar size="1.5em" src={avatar} alt="" className="mr-2" />
          {usernameOrName && `${usernameOrName} `}
          <CaretIcon role="img" aria-hidden focusable="false" />
        </MenuTrigger>

..remaining code..        

      </Menu>
    );
  }

  renderUserMenu() {
    const usernameOrName = getConfig().ENABLE_HEADER_WITHOUT_USERNAME ? this.props.name : this.props.username;
    return this.userMenu(usernameOrName);
  }
  ```
  
  Ofcourse we will update messages file accordingly and will write description for usernameOrName

Reply:
Additionally, Can we please add a comment for this change.

Reply:
Initially, our approach was what you are suggesting and I also agree with this approach but in our initial discussion we decided to create separate components but then we decided that the code would be more redundant, and then we decided to create separate functions.